sdk/core/core-http/lib/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,360 @@
+  }
+
+  const contentElementNames = Object.keys(entry.content).filter(function(key) {
+    return key !== Constants.XML_METADATA_MARKER;
+  });
+
+  if (contentElementNames && contentElementNames[0]) {
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][Constants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[Constants.XML_METADATA_MARKER]) {
+        result[Constants.ATOM_METADATA_MARKER] = entry[Constants.XML_METADATA_MARKER];
+      } else {
+        result[Constants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[Constants.ATOM_METADATA_MARKER]["ContentRootElement"] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== Constants.XML_METADATA_MARKER) {
+          result[Constants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): XMLResponseInJSON[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+/**
+ * Transforms response to contain the parsed data.
+ * @param nameProperties The set of 'name' properties to be constructed on the
+ * resultant object e.g., QueueName, TopicName, etc.
+ * @param response
+ * @param shouldParseResponse
+ */
+export async function deserializeAtomXmlResponse(
+  nameProperties: string[],
+  response: HttpOperationResponse
+): Promise<HttpOperationResponse> {
+  let atomResponseInJson: any;
+  let errorBody: any;
+  try {
+    if (response.bodyAsText) {
+      atomResponseInJson = await deserializeAtomXmlToJson(response.bodyAsText);
+      response.parsedBody = atomResponseInJson;
+    }
+  } catch (e) {
+    errorBody = { code: "ResponseNotInAtomXMLFormat" };
+    return response;
+  }
+
+  // If received data is a non-valid HTTP response, the body is expected to contain error information
+  if (response.status < 200 || response.status >= 300) {
+    errorBody = atomResponseInJson;
+    if (errorBody == undefined) {
+      const HttpResponseCodes = Constants.HttpResponseCodes;
+      const statusCode = response.status;
+      const errorCode = isKnownResponseCode(statusCode)
+        ? HttpResponseCodes[statusCode]
+        : `UnrecognizedHttpResponseStatus: ${statusCode}`;
+      errorBody = {
+        code: errorCode
```

Comment:
Since at this point, we are not dealing with any error body from the service, we dont have to do any normalization. So, I would suggest to throw the Atom Error from here.